imports/plugins/included/product-variant/containers/productsContainer.js

```diff
@@ -0,0 +1,173 @@
+import PropTypes from "prop-types";
+import { Meteor } from "meteor/meteor";
+import { Session } from "meteor/session";
+import { Reaction } from "/client/api";
+import { ITEMS_INCREMENT } from "/client/config/defaults";
+import { ReactionProduct } from "/lib/api";
+import { composeWithTracker } from "/lib/api/compose";
+import { applyProductRevision } from "/lib/api/products";
+import { Products, Tags } from "/lib/collections";
+import ProductsComponent from "../components/products";
+
+/**
+ * loadMoreProducts
+ * @summary whenever #productScrollLimitLoader becomes visible, retrieve more results
+ * this basically runs this:
+ * Session.set('productScrollLimit', Session.get('productScrollLimit') + ITEMS_INCREMENT);
+ * @return {undefined}
+ */
+function loadMoreProducts() {
+  let threshold;
+  const target = document.querySelectorAll("#productScrollLimitLoader");
+  let scrollContainer = document.querySelectorAll("#reactionAppContainer");
+
+  if (scrollContainer.length === 0) {
+    scrollContainer = window;
+  }
+
+  if (target.length) {
+    threshold = scrollContainer[0].scrollTop + scrollContainer[0].offsetHeight - target[0].offsetHeight;
+
+    if (target[0].offsetTop <= threshold) {
+      if (!target[0].getAttribute("visible")) {
+        target[0].setAttribute("productScrollLimit", true);
+        Session.set("productScrollLimit", Session.get("productScrollLimit") + ITEMS_INCREMENT || 24);
+      }
+    } else {
+      if (target[0].getAttribute("visible")) {
+        target[0].setAttribute("visible", false);
+      }
+    }
+  }
+}
+
+class ProductsContainer extends Component {
+
+  static propTypes = {
+    canLoadMoreProducts: PropTypes.bool,
+    initialLoad: PropTypes.bool,
+    products: PropTypes.array,
+    productsSubscription: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      initialLoad: props.initialLoad
+    };
+
+    this.ready = this.ready.bind(this);
+    this.loadMoreProducts = this.loadMoreProducts.bind(this);
+  }
+
+  ready = () => {
+    const isInitialLoad = this.state.initialLoad === true;
+    const isReady = this.props.productsSubscription.ready();
+
+    if (isInitialLoad === false) {
+      return true;
+    }
+
+    if (isReady) {
+      return true;
+    }
+    return false;
+  }
+
+  loadMoreProducts = () => {
+    return this.props.canLoadMoreProducts === true;
+  }
+
+  loadProducts = (event) => {
+    event.preventDefault();
+    this.setState({
+      initialLoad: false
+    });
+    loadMoreProducts();
+  }
+
+  render() {
+    return (
+      <ProductsComponent
+        ready={this.ready}
+        products={this.props.products}
+        productsSubscription={this.props.productsSubscription}
+        loadMoreProducts={this.loadMoreProducts}
+        loadProducts={this.loadProducts}
+      />
+    );
+  }
+}
+function composer(props, onData) {
+  window.prerenderReady = false;
+
+  let stateSlug = "";
+  let stateProducts = [];
```

Comment:
This variable is initialized but then reassigned. Probably one or the other could be removed.